ARCHITECTURE.md

```diff
@@ -0,0 +1,265 @@
+# Solid-client-authn-* architecture
+
+This document summarizes the architecture of the `@inrupt/solid-client-authn-*`
+modules. It applies to both `@inrupt/solid-client-authn-node` and `@inrupt/solid-client-authn-browser`.
+
+## Module map
+
+https://github.com/inrupt/solid-client-authn-js is a Lerna-based monorepo, which
+means this single Git repository actually hosts several npm modules that are
+related to each other. The following diagram shows an overview of the modules and their 
+relationships.
+
+![Module dependencies](./documentation/diagrams/module_map.svg)
+
+`@inrupt/solid-client-authn-node` and `@inrupt/solid-client-authn-browser` (grouped under the
+"Client libraries" label) are the modules we expect developers to import.
+As their names imply, each  of these modules is specific to a given environment 
+(NodeJS or the browser). However, they both have very similar APIs and architecture,
+and mostly differ by their main dependency, namely the third-party library implementing the
+[OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html) protocol.
+
+- `@inrupt/solid-client-authn-node` depends on [`openid-client`](https://github.com/panva/node-openid-client/).
+- `@inrupt/solid-client-authn-browser` depends on `@inrupt/oidc-client-ext`. 
+`@inrupt/oidc-client-ext` in turn depends on
+[`oidc-client`](https://github.com/IdentityModel/oidc-client-js),
+extending it with features we need that it does yet implement, namely support for
+[DPoP tokens](https://tools.ietf.org/html/draft-ietf-oauth-dpop-01) and 
+[Dynamic Client Registration](https://openid.net/specs/openid-connect-registration-1_0.html).
+
+The four modules are available in the standard Lerna [packages directory](./packages).
+
+## OAuth2.0/OpenID Connect
+
+The client libraries aim to help developers authenticating the users of their applications
```

Comment:
maybe (?) help developers authenticate their application users via the OpenID Connect .. ?